query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs

```diff
@@ -374,3 +376,125 @@ pub async fn m2m_disconnect<'conn>(
+        }
+        MongoCommand::Handled { collection, operation } => {
+            let coll = database.collection::<Document>(collection.as_str());
+
+            match operation {
+                MongoOperation::Find(filter, options) => {
+                    let cursor = coll.find_with_session(filter, Some(options), session).await?;
+
+                    cursor_to_json(cursor, session).await?
+                }
+                MongoOperation::FindAndUpdate(filter, modifications, options) => {
+                    let find_one_result = coll
+                        .find_one_and_update_with_session(filter, modifications, options, session)
+                        .await?;
+                    let bson = find_one_result.map(Bson::Document).unwrap_or(Bson::Null);
+                    let json: serde_json::Value = bson.into();
+
+                    json
+                }
+                MongoOperation::FindAndReplace(filter, replacement, options) => {
+                    let find_one_result = coll
+                        .find_one_and_replace_with_session(filter, replacement, options, session)
+                        .await?;
+                    let bson = find_one_result.map(Bson::Document).unwrap_or(Bson::Null);
+                    let json: serde_json::Value = bson.into();
+
+                    json
+                }
+                MongoOperation::FindAndDelete(filter, options) => {
+                    let find_one_result = coll.find_one_and_delete_with_session(filter, options, session).await?;
+                    let bson = find_one_result.map(Bson::Document).unwrap_or(Bson::Null);
+                    let json: serde_json::Value = bson.into();
+
+                    json
+                }
+                MongoOperation::Aggregate(pipeline, options) => {
+                    let cursor = coll.aggregate_with_session(pipeline, options, session).await?;
+
+                    cursor_to_json(cursor, session).await?
+                }
+                MongoOperation::Count(filter, options) => {
+                    let count_res = coll.count_documents_with_session(filter, options, session).await?;
+                    let json: serde_json::Value = count_res.into();
+
+                    json
+                }
+                MongoOperation::Distinct(key, filter, options) => {
+                    let distinct_result = coll.distinct_with_session(key, filter, options, session).await?;
+                    let bson = Bson::Array(distinct_result);
+                    let json: serde_json::Value = bson.into();
+
+                    json
+                }
+                MongoOperation::InsertMany(inserts, options) => {
+                    let insert_result = coll.insert_many_with_session(inserts, options, session).await?;
+                    let json_str = serde_json::to_string(&insert_result)?;
+                    let json = serde_json::from_str(json_str.as_str())?;
+
+                    json
+                }
```

Comment:
This is stupid but I don't see a better way, unfortunately, as `InsertManyResult` doesn't implement `Deserialize`, so I can't convert it to a `serde_json::Value` directly.